File: docs/03-user-guide/examples/python/graph_traverse_prefetch.py
#!/usr/bin/env python3
# REST: Graph traversal with per-call prefetch overrides (Python + requests)

import requests

BASE = "http://localhost:5678"

body = {
    "start_node_id": "n1",
    "max_depth": 3,
    "edge_types": ["REL"],
    "algorithm": "BFS",
}

headers = {
    "Content-Type": "application/json",
    "x-graph-prefetch-enabled": "true",
    "x-graph-prefetch-budget": "8",
}

r = requests.post(f"{BASE}/api/v1/graph/traverse", json=body, headers=headers, timeout=10)
print(r.status_code)
print(r.json())

File: docs/03-user-guide/examples/javascript/graphTraversePrefetch.js
// REST: Graph traversal with per-call prefetch overrides (JavaScript + fetch)

async function traverse() {
  const body = {
    start_node_id: "n1",
    max_depth: 3,
    edge_types: ["REL"],
    algorithm: "BFS",
    // Optional JSON overrides (instead of headers):
    // enable_prefetch: true,
    // prefetch_budget: 8,
  };
  const resp = await fetch("http://localhost:5678/api/v1/graph/traverse", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      // Per-call overrides via headers
      "x-graph-prefetch-enabled": "true",
      "x-graph-prefetch-budget": "8",
    },
    body: JSON.stringify(body),
  });
  console.log(await resp.json());
}

traverse().catch(console.error);

